org.eclipse.scout.rt.mom.api/src/main/java/org/eclipse/scout/rt/mom/api/AbstractMomTransport.java

```diff
@@ -118,12 +132,20 @@ protected Map<Object, Object> lookupEnvironment() {
    *         this call may block until the delegate is initialized).
    */
   protected IMomImplementor getDelegate() {
-    if (!m_delegate.isSet()) {
-      synchronized (m_delegate) {
-        m_delegate.setIfAbsent(this::initDelegate);
+    if (m_delegate == null) {
+      synchronized (m_delegateLock) {
+        if (m_delegate == null) {
+          try {
+            m_delegate = initDelegate();
+          }
+          catch (Exception e) {
+            m_delegate = BEANS.get(NullMomImplementor.class);
+            throw BEANS.get(DefaultRuntimeExceptionTranslator.class).translate(e);
```

Comment:
This change (setting null implmenetor on exception) prevents re-initialization (retry init) for subsequent getDelegate() calls.  Intended behavior?

Reply:
Yes. It should be not retried uncontrolled. After a call to #destroy a re-initialization / retry is possible. #destroy sets m_delegate to null.